src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -1181,10 +1186,12 @@ void RaftReplDev::on_log_found(logstore_seq_num_t lsn, log_buffer buf, void* ctx
     }
 
     rreq->set_lsn(repl_lsn);
+    rreq->set_lentry(lentry);
```

Comment:
This is a bit hacky as we dont set it in normal IO path ,  we believe nuraft will not release them during the whole lifecycle till `commit()`.

I would vote even for normal operations we keep lentry shared_ptr for safety and more consistent behavior.